### ISSUE TITLE
3.16.0 - transaction search doesn't work (0 results always)

### ISSUE THREAD

**Issue by ubeogesh** [braintree/braintree_java#108]
braintreeGateway.transaction().search(new TransactionSearchRequest().orderId().is(orderId)).getFirst()

This works perfectly on 3.15.0 but gives me the following exception on 3.16.0:

    java.lang.IndexOutOfBoundsException: toIndex = 1
	at java.base/java.util.AbstractList.subListRangeCheck(AbstractList.java:507)
	at java.base/java.util.ArrayList.subList(ArrayList.java:1138)
	at com.braintreegateway.ResourceCollection.getFirst(ResourceCollection.java:93)

There is exactly 1 transaction in the search results for 3.15, but on 3.16 the collection is empty

**Comment by jmsjr** [comment]
I second this ... This also happens with search by ID itself, not just by orderID:

```
braintreeGateway.transaction().search(new TransactionSearchRequest().id().is('xxxx'))
```
... where xxxx is a transactionId that was just recently created at Braintree.

Thank goodness for my integration tests. My integration tests ( which were passing with 3.15.0 and now fails with 3.16.0 )  creates transactions in sandbox and then search them.

**Comment by jmsjr** [comment]
Rolled back braintree-java in my Maven POM file to 3.15.0 and my integration tests are working again.

**Comment by jmsjr** [comment]
I found the issue. It is the following change in NodeWrapper in 3.16.0 that broke the behaviour :

https://github.com/braintree/braintree_java/compare/3.15.0...3.16.0#diff-e277bcd14027f28f5bd9ea5a79c2b2c5f37c5ee7a6e2e875dfa3672a38a05cb9

![image](https://user-images.githubusercontent.com/28499925/169263850-91e2d849-63dd-4dcd-9385-c8fdc769a8b9.png)

From the above screenshot, there is nothing returned from `node.getChildren()` and so the `strings` are empty.

**Comment by jmsjr** [comment]
Can we rename this issue to search in general ? Also, given that this method in NodeWrapper is also called by other classes, I presume other behaviours are also now broken :

![image](https://user-images.githubusercontent.com/28499925/169265303-5e55f96f-d977-4498-b101-1a8ece6c23c7.png)

**Comment by crookedneighbor** [comment]
Sorry, we forgot to update here. Yes! We came to the same conclusion on that and have opened a fix internally. We expect the release to go out early next week.

**Comment by hollabaq86** [comment]
Hi folks, we just released 3.16.1 with a fix to this issue. I'm going to close this, please feel free to open a new issue if you encounter any other problems (🙏 ). Thanks for your patience!
